modules/saas.py

```diff
@@ -0,0 +1,2542 @@
+import json
+import re
+import sys
+import time
+import warnings
+from pathlib import Path
+
+import requests
+
+warnings.filterwarnings("always", ".*", PendingDeprecationWarning)
+
+
+class HTTPException(Exception):
```

Comment:
If this Exception is the same as the one in vectra.py, then we may consider moving it to a shared exceptions.py module, or importing directly from vectra.py